src/stream/src/executor/source.rs

```diff
@@ -216,7 +267,9 @@ impl SourceExecutor {
                 .position(|column_id| *column_id == row_id_column_id)
             {
                 let (ops, mut columns, bitmap) = chunk.into_inner();
-                columns[idx] = self.gen_row_column(columns[idx].array().len());
+                if ops.iter().any(|op| *op == Op::Insert) {
```

Comment:
Is this okay? I'm not very clear how we handled delete records from the stream sources before...

Reply:
How about letting the `Source` pass a flag of "whether/where to fill the row_id column" with each stream chunk?

Reply:
This works for stream sources, deleting records from stream source only occurs when pk exist and there will be no row_id.

Reply:
> How about letting the `Source` pass a flag of "whether/where to fill the row_id column" with each stream chunk?

It's a little bit weird too. 🤔

Reply:
Can we just let the Source generate row-ids?

Reply:
Nope, for Delete message the row_id already queried and filled. Generate and refilling is not correct.

Reply:
IMO there will be an append_only flag to SourceExecutor

Reply:
append_only is not helpful for delete/update operation in table source.